app/models/location_report.rb

```diff
@@ -177,6 +177,11 @@ def generate_study_cols_for_row(cur_study)
 
   def search_for_labware_by_selection
     params = { faculty_sponsor_ids:, study_id:, start_date:, end_date:, plate_purpose_ids:, barcodes: }
+    count = Labware.search_for_count_of_labware(params)
+    if count > configatron.fetch(:location_reports_fetch_count_max, 500)
```

Comment:
Is 500 a number discussed with the users?
I would think, as an example, that a standard -80 freezer will hold significantly more than 500 labwares.

Reply:
Maybe find out what location(s) they typically want a report on and check the counts in those.

Reply:
The following shows top 10 biggest reports, (id, count), executed in training console. There are 269,007 lines in the biggest CSV. One can find the distribution of the number of lines to find a sane number but setting a maximum of 500 as the limit does not look right.

```
irb(main):019> top_10 = LocationReport.all.map { |r| [r.id, r.report.read&.split("\n")&.count] }
irb(main):020>                         .reject { |_, count| count.nil? }
irb(main):021>                         .sort_by { |_, count| -count }
irb(main):022>                         .first(10)
=> [[693, 269007], [49, 216651], [1582, 165622], [746, 63515], [1381, 61993], [1451, 61993], [1567, 61867], [1543, 61867], [148, 58952], [341, 58773]]
```